tcmur_cmd_handler.c

```diff
@@ -323,17 +334,24 @@ static int align_and_split_unmap(struct tcmu_device *dev,
 			return TCMU_STS_NO_RESOURCE;
 		}
 
+		urcmd_state = calloc(1, sizeof(*urcmd_state));
+		if (!urcmd_state) {
+			tcmu_dev_err(dev, "Failed to calloc urcmd_state!\n");
+			ret = TCMU_STS_NO_RESOURCE;
+			goto free_desc;
+		}
+			
 		ucmd = calloc(1, sizeof(*ucmd));
 		if (!ucmd) {
 			tcmu_dev_err(dev, "Failed to calloc unmapcmd!\n");
 			ret = TCMU_STS_NO_RESOURCE;
-			goto free_desc;
+			goto free_urcmd;
 		}
 
 		desc->origcmd = origcmd;
 		desc->offset = lba * block_size;
 		desc->length = lbas * block_size;
-		ucmd->cmdstate = desc;
+		urcmd_state->cmdstate = desc;
```

Comment:
Here you didn't set the urcmd_state to ucmd.

Reply:
Good catch! Will fix that. Thx.